packages/block-editor/src/store/reducer.js

```diff
@@ -1223,6 +1223,34 @@ export function lastBlockAttributesChange( state, action ) {
 	return null;
 }
 
+/**
+ * Reducer returning an array of discover blocks.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export const discoverBlocks = ( state = { results: {}, hasPermission: undefined, filterValue: undefined }, action ) => {
```

Comment:
I imagine one question mark will be whether this should be in the `block-editor` store. Or whether any of this code should be part of `block-editor`. 

I could imagine it being its own `discover-blocks` package with its own store and rendering through a `Slot`. It might be interesting to see what the opinion is from more experienced contributors about where this should live.

Reply:
That is true. It doesn't feel good to add more to a store that is already quite big. 
Other considerations for a single `block-editor` store and separate `discover-blocks` package/store:
-  There are also other parts/functionalities that are not started yet that might give us a clearer view on this.
- - Installing/uninstalling a block in the server from block editor.
- - Extending sidebar to display a list of "blocks to install".
- To a user, the inserter's search is a single search for all the blocks: core, installed, reusable and discover. But currently the implementation is separated on the front end, and there are some discussion about combining the search on server side.

Reply:
I didn't see this comment earlier. It is exactly the same concern I had when commenting here: https://github.com/WordPress/gutenberg/pull/16524#issuecomment-515904041.